File: Array&String/CPP/positions_of_large_groups.cpp
#include <vector>
#include <string>
using namespace std;

class Solution {
public:
    vector<vector<int>> largeGroupPositions(string s) {
        int start = 0, end = 0, max = 2;
        vector<vector<int>> maxIdx;
        char cur = s[0];
        for (int i = 0; i < s.size(); i++) {
            if (s[i] != cur) {
                if (end-start >= max) maxIdx.push_back(vector<int>{start, end});
                end = start = i;
                cur = s[i];
                continue;
            } end = i;
        }
        if (start != end && end - start >= max) maxIdx.push_back(vector<int>{start, end});
        return maxIdx;
    }
};

File: Array&String/positions_of_large_groups.py
import re

class Solution:
    def largeGroupPositions(self, s: str) -> list[list[int]]:
        start = end = 0
        cur, ans, maxe = s[0], [], 2
        for i, e in enumerate(s):
            if e != cur:
                if end - start >= maxe:
                    ans.append([start, end])
                start = end = i
                cur = e
            else:
                end = i
        if start != end and end - start >= maxe:
                    ans.append([start, end])
        return ans
    
    # shorter 
    def largeGroupPositions(self, S: str) -> list[list[int]]:
        i, j, N = 0, 0, len(S)
        res = []
        while i < N:
            while j < N and S[j] == S[i]: j += 1
            if j - i >= 3: res.append([i, j - 1])
            i = j
        return res
    
    # using regex
    def largeGroupPositions(self, S: str) -> list[list[int]]:
        return [[r.start(), r.end() - 1] for r in re.finditer(r'(\w)\1{2,}', S)]